### PR TITLE
fix: remove unecessary calls to third party apis

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

The CollectibleContracts component was making more calls to the NFT API than it should.

Users could either have the nft detection enabled or disabled.

When the nft detection is enabled, the component was:
1- Executing `detectNfts` function which triggers one call to get all nfts for a user
2- Calling `updateNftMetadata` which triggers `N` number of calls to the NFT-API assuming the user has `N` NFT.

The second call is unnecessary if the user already have the nftDetection enabled.

In this fix, i added a check in the useEffect to see if the user has the nftDetection enabled or not

```
    if (updatableCollectibles.length !== 0 && !useNftDetection) {
      updateAllCollectibleMetadata(updatableCollectibles);
    }
```
If it is already enabled, calling the detectNfts function should be enough to refresh metadata.

When the user has the nftDetection Disabled, in this case the `updateNftMetadata` will be executed. 
I made an update in the `updateNftMetadata` fct to not execute the calls every time the user pulls down t refresh. A 10 mins interval should be enough.

Note: We will have an update on the nftDetectionController in the future to update the polling strategy.

## **Related issues**

- Compare url on core: https://github.com/MetaMask/core/compare/patch/mobile-assets-controllers-26...patch/mobile-assets-controllers-26-optimize-nft-refresh-calls?expand=1
- https://github.com/MetaMask/core/pull/4325

## **Manual testing steps**

To check the unecessary calls:

1. Build on main
2. Go to app/components/UI/CollectibleContracts/index.js and add a log inside useEffect
3. Go to node-modules, nftDetectionController.js function getOwnerNfts and add a console.log
4. Go to node-modules, nftController.js function getNftInformationFromApi and add a log
5. When you have nft detection disabled, go to nft tab, and pull down to refresh and check the calls made in your terminal
6. Enable nft detection, go to nft tab, pull down to refresh and check your logs

(See before video, notice on that wallet i have 5 nfts, it should be making only 1 call per nft => 5 calls in total but its making more than that. Every state update will result in the useEffect being executed which we attempt to avoid in this fix)

To check the update on this PR:

1- Build on this branch
2. Go to app/components/UI/CollectibleContracts/index.js and add a log inside useEffect
3. Go to node-modules, nftDetectionController.js function getOwnerNfts and add a console.log
4. Go to node-modules, nftController.js function getNftInformationFromApi and add a log
5. When you have nft detection disabled, go to nft tab, and pull down to refresh and check the calls made in your terminal
6. Enable nft detection, go to nft tab, pull down to refresh and check your logs

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/10994169/77f7ba9e-add9-4ce8-ad78-bc41926276b9

### **After**

https://github.com/MetaMask/metamask-mobile/assets/10994169/4677a899-9fdd-46a3-98e3-9422d6fa9eb9

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by legobeat** [comment]
Great catch!
1. The `patches/@metamask+assets-controllers+26.0.0.patch` diff is big. Do you have a reference for the actual intended commit(s) on https://github.com/MetaMask/core?
2. Would it be possible to construct a test or two to verify correct behavior and prevent future regressions?

**Comment by sahar-fehri** [comment]
> Great catch!
> 
> 1. The `patches/@metamask+assets-controllers+26.0.0.patch` diff is big. Do you have a reference for the actual intended commit(s) on https://github.com/MetaMask/core?
> 2. Would it be possible to construct a test or two to verify correct behavior and prevent future regressions?

Hey @legobeat ! I have this core PR for these updates https://github.com/MetaMask/core/pull/4325
Will work on the unit tests 🙏

File: app/components/UI/CollectibleContracts/index.js [hunk 1]
```diff
@@ -356,6 +330,7 @@ const CollectibleContracts = ({
         data={collectibleContracts}
         renderItem={({ item, index }) => renderCollectibleContract(item, index)}
         keyExtractor={(_, index) => index.toString()}
+        testID="refreshControl"
```
Comment (tommasini) [inline app/components/UI/CollectibleContracts/index.js:333]: Can we move this testId to a constant? I would create a new file in the component folder to store those, something like "CollectibleContracts.constants.ts"

**Review by tommasini — APPROVED** [review]
LGTM!

**Pull request merged**